app/main/posts/views/post-view-map.directive.js

```diff
@@ -85,6 +88,7 @@ function PostViewMap(PostEndpoint, Maps, _, PostFilters, L, $q, $rootScope, $com
                 return $scope.filters;
             }, function (newValue, oldValue) {
                 if (newValue !== oldValue) {
+                    clearData();
                     reloadMapPosts();
```

Comment:
We need to make sure we show some kind of `loading` status in the UI while this happens now.